contracts/RMRK/extension/tokenProperties/RMRKTokenPropertiesRepository.sol

```diff
@@ -0,0 +1,471 @@
+// SPDX-License-Identifier: Apache-2.0
+
+pragma solidity ^0.8.18;
+
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "@openzeppelin/contracts/token/ERC721/IERC721.sol";
+import "@openzeppelin/contracts/utils/introspection/IERC165.sol";
+
+import "../../library/RMRKErrors.sol";
+import "./IRMRKTokenPropertiesRepository.sol";
+
+/**
+ * @title RMRKTokenPropertiesRepository
+ * @author RMRK team
+ * @notice Smart contract of the RMRK Token property repository module.
+ */
+contract RMRKTokenPropertiesRepository is IRMRKTokenPropertiesRepository {
+    mapping(address => mapping(uint256 => AccessType))
+        private _parameterAccessType;
+    mapping(address => mapping(uint256 => address))
+        private _parameterSpecificAddress;
+    mapping(address => IssuerSetting) private _issuerSettings;
+    mapping(address => mapping(address => bool)) private _collaborators;
+
+    // For keys, we use a mapping from strings to IDs.
+    // The purpose is to store unique string keys only once, since they are more expensive,
+    mapping(address => mapping(string => uint256)) private _keysToIds;
+    mapping(address => uint256) private _totalProperties;
+
+    // For strings, we also use a mapping from strings to IDs, together with a reverse mapping
+    // The purpose is to store unique string values only once, since they are more expensive,
+    // and storing only IDs.
+    mapping(address => uint256) private _totalStringValues;
+    mapping(address => mapping(string => uint256)) private _stringValueToId;
+    mapping(address => mapping(uint256 => string)) private _stringIdToValue;
+    mapping(address => mapping(uint256 => mapping(uint256 => uint256)))
+        private _stringValueIds;
+
+    mapping(address => mapping(uint256 => mapping(uint256 => address)))
+        private _addressValues;
+    mapping(address => mapping(uint256 => mapping(uint256 => bytes)))
+        private _bytesValues;
+    mapping(address => mapping(uint256 => mapping(uint256 => uint256)))
+        private _uintValues;
+    mapping(address => mapping(uint256 => mapping(uint256 => bool)))
+        private _boolValues;
+
+    struct IssuerSetting {
+        bool registered;
+        bool useOwnable;
+        address issuer;
+    }
+
+    /**
+     * @inheritdoc IRMRKTokenPropertiesRepository
+     */
+    function registerAccessControl(
+        address collection,
+        address issuer,
+        bool useOwnable
```

Comment:
I thought we agreed on removing this :grimacing:  Besides risky, it makes the code more complex on a few places
https://github.com/rmrk-team/evm/pull/186#discussion_r1161619697

Reply:
I thought we agreed only on removing the ability to use the collection that doesn't have ownable implemented 🤔

Reply:
All we care is that it has owner() method, even if they don't claim support the interface.
If they have it good, owner() we can register properties. If not, sorry for them :man_shrugging:

Reply:
Yeah, this implementation does just that. You don't need to use ownable when using it, but you need the `owner` method in order to register access control.